lib/netext/httpext/request.go

```diff
@@ -269,8 +270,24 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 	}
 
 	if preq.Auth == "digest" {
+		// In both digest and NTLM it is expected that the first response will be 401
+		// this does mean that a non 401 response will be marked as failed/unexpected
+		if tracerTransport.responseCallback != nil {
+			originalResponseCallback := tracerTransport.responseCallback
+			tracerTransport.responseCallback = func(status int) bool {
+				tracerTransport.responseCallback = originalResponseCallback
+				return status == 401
+			}
+		}
 		transport = digestTransport{originalTransport: transport}
 	} else if preq.Auth == "ntlm" {
+		if tracerTransport.responseCallback != nil {
+			originalResponseCallback := tracerTransport.responseCallback
+			tracerTransport.responseCallback = func(status int) bool {
+				tracerTransport.responseCallback = originalResponseCallback
+				return status == 401
```

Comment:
Hmm I'm not sure this is correct. I think it should be something like `status == 401 || originalResponseCallback(status)`, considering that NTLM is a connection-level authentication IIRC. So, it's possible for the first request to actually pass through completely fine without any authentication, if it was reusing a connection that was previously authenticated.

Something similar will be true for digest auth eventually, when we get around to https://github.com/loadimpact/k6/issues/800, but as it is should be fine for now, I think...

Reply:
hm ... I guess you are right, will change it, but I have no way of testing it so :shrug: